### ISSUE TITLE
Clang static analyzer

### ISSUE THREAD

**Issue by cyberkm** [fastbuild/fastbuild#136]
I just wondering, if there is a way to somehow make fastbuild like clang static analyzer.

I tried to add make a rule to compile code with --analize parameter, but it seems that --analyze doesn't like -E flag that fastbuild adds.

**Comment by ffulin** [comment]
I'm not too familiar with Clang's static analysis, but it'd be great to support it.  (Distributed static analysis maybe? :) )

FASTBuild splits compilation into two passes when using Clang.  The first pass is just pre-processing, and that's done with the /E option. It makes sense that this would be incompatible with static analysis since no compilation is taking place.  FASTBuild could be modified to strip the analyze command line option when it's doing this first pass, leaving it on for the second pass (the actual compilation). That might be all that's needed?

**Comment by nil-ableton** [comment]
One way to do this would be for the user to explicitely state `PreProcessorOptions` in say, an `ObjectList`

And then only afterwards add the --analyze flag to CompilerOptions

like so:
```
.Preprocessor = '$Compiler$'
.PreprocessorOptions = '$CompilerOptions$'
.CompilerOptions + ' --analyze'
```

**Comment by cyberkm** [comment]
Adding compiler flag later created too much maintenance for me in bff file.

So I created a pull request for this issue
https://github.com/fastbuild/fastbuild/pull/137

**Comment by ffulin** [comment]
I think having FASTBuild handle this arg automatically during preprocessing (as per your pull request) is the correct approach to simplify things for the user.

**Comment by ffulin** [comment]
Thanks to your changes, a fix for this will be in the next release (v0.92). I'll keep this issue open until that version is released. Thanks again!

**Comment by ffulin** [comment]
v0.92 has now been released and contains a fix for this. Thanks again for reporting/helping to fix this issue!
